store.go

```diff
@@ -413,15 +414,32 @@ func (ls *LimitStore) Collect(id SpanID, anns ...Annotation) error {
 	ls.mu.Lock()
```

Comment:
Better to add a `defer ls.mu.Unlock()` here, and delete all the calls to `ls.mu.Unlock()` below